libmscore/key.cpp

```diff
@@ -105,15 +106,23 @@ bool KeySigEvent::operator==(const KeySigEvent& e) const
 //   transposeKey
 //---------------------------------------------------------
 
-Key transposeKey(Key key, const Interval& interval)
+Key transposeKey(Key key, const Interval& interval, PreferSharpFlat prefer)
       {
       int tpc = int(key) + 14;
       tpc     = transposeTpc(tpc, interval, false);
+
+      // change between 5/6/7 sharps and 7/6/5 flats
+      if (tpc <= 9 && prefer == PreferSharpFlat::SHARPS)
+            tpc += 12;
+      if (tpc >= 19 && tpc <= 21 && prefer == PreferSharpFlat::FLATS)
+            tpc -= 12;
```

Comment:
After thinking for a while I guess I understand now why did you put those 9, 19 and 21 in conditions (I guess, because otherwise enharmonic change will make a key with more than 7 sharps/flats?), but could you please add a comment here to make this idea more clear?

Reply:
Done.